src/coreclr/jit/optimizer.cpp

```diff
@@ -8898,6 +8898,581 @@ bool Compiler::optIsRangeCheckRemovable(GenTree* tree)
     return true;
 }
 
+/*
+    Optimizes boolean when the second BB is BBJ_COND
```

Comment:
Please describe in more detail what kind of optimization this function does. E.g., what is the transformation, what patterns are handled, what are the limitations?